helios-services/src/main/java/com/spotify/helios/rollingupdate/RollingUpdateOpFactory.java

```diff
@@ -103,41 +120,49 @@ public RollingUpdateOp yield() {
 
-    final DeploymentGroupStatus status = DeploymentGroupStatus.newBuilder()
-        .setState(FAILED)
-        .setError(errMsg)
-        .build();
+    if (isOverFailureThreshold()) {
+      final DeploymentGroupStatus status = DeploymentGroupStatus.newBuilder()
+          .setState(FAILED)
+          .setError(errMsg)
+          .build();
 
-    // Delete tasks, set state to FAILED
-    operations.add(delete(Paths.statusDeploymentGroupTasks(deploymentGroup.getName())));
-    operations.add(set(Paths.statusDeploymentGroup(deploymentGroup.getName()), status));
+      // Delete tasks, set state to FAILED
+      operations.add(delete(Paths.statusDeploymentGroupTasks(deploymentGroup.getName())));
+      operations.add(set(Paths.statusDeploymentGroup(deploymentGroup.getName()), status));
 
-    final RolloutTask task = tasks.getRolloutTasks().get(tasks.getTaskIndex());
+      // Emit a FAILED event and a failed task event
+      final Map<String, Object> taskEv = eventFactory.rollingUpdateTaskFailed(
+          deploymentGroup, task, errMsg, errorCode, metadata);
+      events.add(taskEv);
+      events.add(eventFactory.rollingUpdateFailed(deploymentGroup, taskEv));
+    } else {
+      return nextTask(Collections.<ZooKeeperOperation>emptyList(), task.getTarget(), errMsg,
+                      errorCode);
+    }
 
-    // Emit a FAILED event and a failed task event
-    final List<Map<String, Object>> events = Lists.newArrayList();
-    final Map<String, Object> taskEv = eventFactory.rollingUpdateTaskFailed(
-        deploymentGroup, task, errMsg, errorCode, metadata);
-    events.add(taskEv);
-    events.add(eventFactory.rollingUpdateFailed(deploymentGroup, taskEv));
+    return new RollingUpdateOp(ImmutableList.copyOf(operations), ImmutableList.copyOf(events));
+  }
 
-    return new RollingUpdateOp(ImmutableList.copyOf(operations),
-                               ImmutableList.copyOf(events));
+  public RollingUpdateOp taskError(final String msg, final String host,
+                                   final RollingUpdateError errorCode) {
+    return taskError(msg, host, errorCode, Collections.<String, Object>emptyMap());
   }
 
-  public RollingUpdateOp error(final String msg, final String host,
-                               final RollingUpdateError errorCode) {
-    return error(msg, host, errorCode, Collections.<String, Object>emptyMap());
+  public RollingUpdateOp taskError(final Exception e, final String host,
+                                   final RollingUpdateError errorCode) {
+    return taskError(e.getMessage(), host, errorCode);
   }
 
-  public RollingUpdateOp error(final Exception e, final String host,
-                               final RollingUpdateError errorCode) {
-    return error(e.getMessage(), host, errorCode);
+  private boolean isOverFailureThreshold() {
+    return ((float) (tasks.getFailedTargets().size() + 1) / tasks.getNumTargets() * 100)
```

Comment:
Maybe move the failed target count to be a param, so that if somebody calls `isOverFailureThreshold` in the future they aren't confused why it always counts at least one host as failed.